### PR TITLE
Generate frontend server's .packages from Dart .packages file instead of using pub get.

### PR DESCRIPTION
**Pull request by mraleph**
Dart SDK itself does not use pub which forced us to manually maintain dependecy
overrides in the frontend server's pubspec.yaml. This is rather fragile and
easily breaks when dependencies in one of the "internal" packages change, but
the package is not published yet.

This change switches us to the same model of dependcies management that Dart SDK
itself is using.

Down side of this is that we can no longer do `pub run test` in the
frontend_server and have to run tests manually.

### CODE REVIEW

File: tools/frontend_server_packages.py [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+# Dart SDKs .packages file located in third_party/dart/.packages
+
+import os
+import shutil
+
+FRONTEND_SERVER_DIR = os.getcwd()
+SRC_DIR = os.path.dirname(os.path.dirname(FRONTEND_SERVER_DIR))
+DART_PACKAGES_FILE = os.path.join(SRC_DIR, 'third_party', 'dart', '.packages')
+
+with open('.package', 'w') as packages:
```
Comment (aam) [inline tools/frontend_server_packages.py]: .packages?

File: DEPS [hunk 2]
```diff
@@ -88,6 +88,7 @@ vars = {
   'dart_stack_trace_tag': '1.9.2',
   'dart_stream_channel_tag': '1.6.4',
   'dart_string_scanner_tag': '1.0.2',
+  'dart_term_glyph_tag': '1.0.0',
```
Comment (aam) [inline DEPS]: Was this change intentional? It seems to be unrelated to the generation of .packages file.

File: tools/frontend_server_packages.py (hunk 1 above)
Comment (mraleph) [reply tools/frontend_server_packages.py]: I can't name a variable `.packages`.

File: DEPS (hunk 2 above)
Comment (mraleph) [reply DEPS]: Yes. This is a package needed to run the test. (I think test package depends on it).

It used to be pulled by `pub get`.

Comment (aam) [reply DEPS]: Ah, interesting.

**Comment by mraleph** [comment]
@aam I have fixed the python script :) previously things were passing because I had a stale `.packages`. Now I am using the one generated by the python script and things work.

**Comment by aam** [comment]
Sounds good! Thanks for fixing this!

**Comment by mraleph** [comment]
I think the `flutter_kernel_transformers` would need a similar fix.

**Comment by aam** [comment]
cc @jacob314 then

**Comment by aam** [comment]
> I think the flutter_kernel_transformers would need a similar fix.

Yes, this needs to be done now to unblock dart->flutter engine roll.
